ecl/hqlcpp/hqlhtcpp.cpp

```diff
@@ -14119,43 +14134,51 @@ ABoundActivity * HqlCppTranslator::doBuildActivityDedup(BuildCtx & ctx, IHqlExpr
         //virtual unsigned recordToKey(void * _key, const void * _record)
         buildDedupSerializeFunction(instance->startctx, "recordToKey", dataset, keyDataset, info.equalities, selects, selSeq);
 
-        //virtual ICompare * queryKeyCompare()
-        bool reuseCompare = false;
-        OwnedHqlExpr keyOrder = createValueSafe(no_sortlist, makeSortListType(NULL), selects);
-        if (recordTypesMatch(dataset, keyDataset))
+        if (info.keepBest)
```

Comment:
That's correct.  With KeepBest the entire row is kept in the hash table.... I've added a comment in the source code, explaining this.